python/lsst/meas/extensions/gaap/gaap.py

```diff
@@ -0,0 +1,186 @@
+            baseName = self.getGaapResultName(sF, sigma)
+            schema.addField(schema.join(baseName, "instFlux"), type="F", doc="GAaP Flux")
+            schema.addField(schema.join(baseName, "instFluxErr"), type="F", doc="GAaP Flux error")
+
+    def convolve(self, exposure: afwImage.Exposure, modelPsf: afwDetection.GaussianPsf,
+                 footprint: afwDetection.Footprint) -> afwImage.Exposure:
+        """Convolve the ``exposure`` to make the PSF same as ``modelPsf``.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Original (full) exposure containing all the sources.
+        modelPsf : `lsst.afw.detection.GaussianPsf`
+            Target PSF to which to match.
+        footprint : `lsst.afw.detection.Footprint`
+            Footprint of the source to measure.
+
+        Returns
+        -------
+        convExp : `lsst.afw.image.Exposure`
+            Sub-image containing the source, convolving to the target seeing
+        """
+
+        bbox = footprint.getBBox()
+        pixToGrow = self.config.modelPsfMatch.kernel.active.detectionConfig.fpGrowPix
+        # This is conservative for now.
+        # An alternate option is 2*max(self.psfMatch.kConfig.sizeCellX, self.psfMatch.kConfig.sizeCellY)
+        bbox.grow(pixToGrow)  # gets too big
+
+        origPsf = exposure.getPsf()
+
+        maskedImage = exposure.getMaskedImage()
+        subImage = maskedImage.Factory(maskedImage, bbox)
+        subExposure = afwImage.ExposureF(subImage)
+        subExposure.setPsf(origPsf)
+
+        config = self.config.modelPsfMatch
+        config.setDefaults()
+        config.kernel.active.alardSigGauss = [modelPsf.getSigma()]  # The size has to be set dynamically
+        task = config.target(config=config)
+        result = task.run(exposure=subExposure, referencePsfModel=modelPsf)
+        # TODO: DM-27407 will re-Gaussianize the exposure to make the PSF even more Gaussian-like
+        convolved = result.psfMatchedExposure
+        # Setting nans to zeros in a hacky way! DM-28740 will fix this properly.
+        convolved.image.array[np.isnan(convolved.image.array)] = 0.0
+        return convolved
+
+    def measure(self, measRecord: lsst.afw.table.SourceRecord, exposure: afwImage.Exposure,
+                center: lsst.geom.Point2D) -> None:
+        """Measure the GAaP flux
+        """
+
+        psf = exposure.getPsf()
+        if psf is None:
+            raise lsst.meas.base.MeasurementError("No PSF in exposure")
+
+        seeing = psf.computeShape(center).getTraceRadius()
+        for sF in self.config.scalingFactors:
+            if sF < 1.0:
+                # Do not allow the PSF matching to go into deconvolution mode.
+                raise lsst.meas.base.FatalAlgorithmError("The scaling factor has to be greater than"
+                                                         "or equal to 1")
```

Comment:
The ``itemCheck`` keyword argument seems to be the thing to do here, although it'd mean that the measurement plugin will not be run as opposed to just not running the measurement for invalid values. That behavior should still be fine, I suppose?

Reply:
Yeah, I think if a configuration option is sufficiently wrong that some aspect of the measurement will never be run (rather than just not run on certain kinds of data), it's generally better to fail completely and early.  It's probably possible to find counterexamples, but I think it's a good principle.